paasta_tools/firewall_update.py

```diff
@@ -12,6 +12,7 @@
 from inotify.adapters import Inotify
```

Comment:
All this code looks reasonable to me, but some thoughts about the approach of not allowing access to backends directly if a proxy_port exists:

1. As written, it's going to be pretty much impossible for a service to transition from accessing backends directly to using the proxy_port. As soon as they add a proxy_port, the firewall stops allowing access to the backends. They really need a transition period where both are allowed so that they can go update the clients to access the proxy_port before disabling direct access to the backends.

2.  @bobtfish mentioned in PAASTA-11261 that some services have a proxy_port which is used for the initial request, but afterwards they talk to the backends directly (he mentioned Vault and ES).

I think some kind of toggle for allowing direct access to the backends even when there's a proxy_port could fix this?